src/coreclr/nativeaot/System.Private.CoreLib/src/System/CrashInfo.cs

```diff
@@ -172,7 +172,8 @@ private bool WriteExceptionHelper(ReadOnlySpan<byte> key, Exception exception, i
             if (!OpenValue(key, '{'))
                 return false;
 
-            if (!WriteHexValue("address"u8, (ulong)Unsafe.AsPointer(ref exception)))
+            ulong address = Unsafe.As<Exception, nuint>(ref exception);
```

Comment:
Can the GC move the exception object between when we write this address in crashinfo and when the actual dump is generated?

Reply:
Yes, it can. IIRC, it was discussed somewhere. It is not 100% reliable pattern.

The diagnostic stack likes to use it. Tracing uses in multiple places, e.g. https://github.com/dotnet/runtime/blob/9fa5128e2d16855bef83560a5a38b493fa3eb7ea/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TplEventSource.cs#L505 .